### PR TITLE
[ufw_firewall] add VM firewall rules

### LINKED ISSUE #4965: automate local firewall rules for vms in the .lib domain
### What maintenance needs to be done?

Create firewall rules for all our virtual machines. We do not use PanOS on our [*.lib.* domain](https://datatracker.ietf.org/doc/html/rfc1918) and therefor will need to add the same rules on the host firewall. 

### Level of urgency
- [ ] High
- [ ] Moderate
- [x] Low

### Why is this maintenance needed?
- While the non-routable ip addresses are only reachable to us having firewall rules is probably a good idea across all our infrastructure

### Acceptance criteria
 - [ ] A UFW role
 - [ ] Variables for the ports to open

### PR DESCRIPTION
**Pull request by kayiwa**
- **a parametrized role for a hosts firewall**
- **add documentation for the role**
- **our lowest version is no longer bionic**
- **add ufw_firewall to roles to be tested**

closes #4965

### CODE REVIEW

File: roles/firewall/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+# defaults file for roles/ufw_firewall
+firewall_rules:
```
Comment (kayiwa) [inline roles/firewall/defaults/main.yml:3]: add ufw as variable to avoid confusing it with panos

File: roles/firewall/handlers/main.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+# handlers file for roles/ufw_firewall
+- name: restart ufw
+  ansible.builtin.service:
+    name: ufw
+    state: restarted
+
+- name: reload firewalld
```
Comment (kayiwa) [inline roles/firewall/handlers/main.yml:8]: we won't be using firewalld

File: roles/firewall/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+---
+# tasks file for roles/ufw_firewall
+- name: Firewall | Determine OS family
```
Comment (kayiwa) [inline roles/firewall/tasks/main.yml:3]: we are using ufw

**Review by carolyncole — APPROVED** [review]
We reviewed in Open house on 10/28

**Pull request merged**